### PR TITLE
[mesheryctl] fall back to last release when patch release tar ball is not available

### LINKED ISSUE #1573: meshery install script fails with patch releases
### Description
`curl -L https://meshery.io/install | bash - ` fails when the latest release is of type patch, for e.g. v0.7.3-patch.3

### Expected Behavior
install script should not fail, when the mesheryctl install packages are not available under:
https://github.com/meshery/meshery/releases/, by falling back to last stable version
Rather, the mesheryctl executables should made available for patch/beta/alpha/rc releases.
https://github.com/meshery/meshery/pull/9642 talks about the recent changes to the publishing of the install packages.

### Screenshots
image (10)

### Enviroment:
 - OS: Any
 - Browser: Any
 - Version: v0.7.3-patch.3
 - Device: Desktop Mobile

---
## Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

The meshery.io website uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/meshery/meshery.io/tree/master).
- 📚 See the [Contributing to Meshery.io Website](https://github.com/meshery/meshery.io#contributing-to-the-mesheryio-website) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by hexxdump**
**Description**
Mesheryctl install script should fetch the last available release's installables,
when the requested patch release's installables are not available in the release page.
This PR fixes #1573 

**Notes for Reviewers**
Logic is tested locally with debug logs.

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [x ] Yes, I signed my commits.
 Signed-off-by: Hussaina Begum Nandyala [hexxdump@gmail.com](mailto:hexxdump@gmail.com)

### CODE REVIEW

File: install [hunk 1]
```diff
@@ -123,16 +123,21 @@ if ! command_exists tar; then
...
-                  awk -F '/' '{print $NF}' )
+  MESHERY_VERSION=$(curl -L -s https://docs.meshery.io/project/releases/latest)
 fi
 
 if [ "x${MESHERY_VERSION}" = "x" ] ; then
   printf "Unable to get latest mesheryctl version. Set MESHERY_VERSION env var and re-run. For example: export MESHERY_VERSION=v0.1.6\n"
   exit;
 fi
 
+# For patch releases, the meshery tar ball will not be generated, so fall back to last stable release
```
Comment (nishantwrp) [inline install:134]: I see other type of releases as well, e.g. `beta`, `rc`. Do we not need special handling for those cases?

**Review by nishantwrp — APPROVED** [review]

File: install (hunk 1 above)
Comment (hexxdump) [reply install:134]: @nishantwrp, for beta and rc releases, tar balls are available.
For e.g. 
https://github.com/meshery/meshery/releases/tag/v0.7.0-beta-4
https://github.com/meshery/meshery/releases/tag/v0.6.0-rc-2
https://github.com/meshery/meshery/releases/tag/v0.6.0-rc-5j

**Comment by hexxdump** [comment]
@leecalcote, let me know if this can be merged?

**Review by leecalcote — APPROVED** [review]
@hexxdump I missed this... yes, this looks great. 👍

**Pull request merged**